apps/system/js/modal_dialog.js

```diff
@@ -362,23 +362,24 @@ var ModalDialog = {
 
   buildSelectOneDialog: function md_buildSelectOneDialog(data) {
     var elements = this.elements;
-    elements.selectOneTitle.textContent = data.title;
-    elements.selectOneMenu.innerHTML = '';
+    this._localizeElement(elements.selectOneTitle, data.title);
```

Comment:
With your refactor, you're changing:

    elements.selectOneTitle.textContent = data.title

to

    elements.selectOneTitle.setAttribute('data-l10n-id', data.title);

I'm not sure what this does, was it on purpose or did you intend to write the following instead?

    this._localizeElement(elements.selectOneTitle, {raw: data.title});